### DISCUSSION TITLE
Best way to initialize a fractional fixed point value.

### DISCUSSION THREAD

**Discussion by joaomontenegro** [PaulRBerg/prb-math#45] [Q&A] [answered]
What's the best way to initialize a uint256 with a fractional number?

Pseudocode of an example of what I am trying to achieve:
`uint256 x = 0.123;`

**Comment by PaulRBerg** [comment, marked as answer]
Hey! There are multiple ways to do this, as there is a multitude of [conversion function](https://github.com/paulrberg/prb-math/tree/b6066c8794ff2eaa6ccf667612fbe4bddad8a9a3#conversion-functions) shipped with PRBmath.

### Scientific Notation

```solidity
// 0.5
UD60x18 foo = ud(0.5e18);
UD60x18 foo = ud60x18(0.5e18);
UD60x18 foo = wrap(0.5e18);
```

### "To" Conversion Function

This only works for whole numbers greater than or equal to 1e18. See the NatSpec comments for the [`toUD60x18`](https://github.com/paulrberg/prb-math/blob/b6066c8794ff2eaa6ccf667612fbe4bddad8a9a3/src/UD60x18.sol#L582-L589) for more details on how this works.

```solidity
// 50
UD60x18 foo = toUD60x18(50);
```
